### ISSUE TITLE
Create tables to utilize dynamic grouping from left to right of content

### ISSUE THREAD

**Issue by poloka** [cerner/terra-core#3544]
# Feature Request
Requesting a solution for building a table where content is grouped from left to right in a dynamic manner.  See the following article on the example:
https://stackoverflow.com/questions/38330484/dynamic-complex-rowspan-in-html-table

## Description
The feature would be to allow creation of a table that can dynamically group content from left to right that are shared across the various rows.  The assumption would be that there would be an identifier added to each `headerData` element indicating it should be a grouper.  An additional consideration should be that if there is a column that does not contain any content, provide the option to the consumer to indicate the column could be automatically removed to save space.  From this, a consumer can create multiple rows for the `bodyData` as if a standard `Table` and allow the Terra component to handle the construction of the table.

**Comment by poloka** [comment]
Per discussion with @benbcai closing request and logged a JIRA for the request.
